### PR TITLE
When m_max_sessions is zero, don't prune the session cache.

### PR DESCRIPTION
**Pull request by oviano**
Fixes https://github.com/randombit/botan/issues/3438.

### CODE REVIEW

**Review by randombit — COMMENTED** [review]
Seems fine to me. @reneme?

File: src/lib/tls/sessions_sql/tls_session_manager_sql.cpp [hunk 1]
```diff
@@ -296,6 +296,9 @@ void Session_Manager_SQL::prune_session_cache()
    {
    // internal API: assuming that the lock is held already
 
+   if(!m_max_sessions)
+      return;
```
Comment (reneme) [inline src/lib/tls/sessions_sql/tls_session_manager_sql.cpp:300]: For clarity, maybe:

```suggestion
   if(m_max_sessions == 0)
      return;
```

**Review by reneme — CHANGES_REQUESTED** [review]
It would be great to add a test in `test_tls_session_manager.cpp`. More specifically add a `CHECK` section in the `test_session_manager_sqlite()` function as a regression test.

File: src/lib/tls/sessions_sql/tls_session_manager_sql.cpp (hunk 1 above)
Comment (oviano) [reply src/lib/tls/sessions_sql/tls_session_manager_sql.cpp]: I'm not familiar with how the tests work/what exactly you want it to test, so I'll have to leave that to somebody else.

**Comment by reneme** [comment]
I'll add the test in a follow-up. Thanks @oviano.

**Review by reneme — APPROVED** [review]

**Pull request merged**